tests/Type/data/conditionable.php

```diff
@@ -24,6 +24,12 @@ class Foo
     return rand();
 }));
 
+/** @var string|null */
+$var = 'string'
+(new Foo())->when($var, function (Foo $foo, string $var) {
```

Comment:
I think this test would always pass because of the `string` typehint here. Can you remove it and see if still passes?

Reply:
Done!

Reply:
Yeah, as you can see tests are failing now. So this is not the correct way to handle this I guess.

Reply:
Yeah, this test doesn't work because PHPStan will ignore the `@var` annotation and still think it's a `string`. In the PR I did before to support `transform` I used `User::first()` which is definitely `App\User|null`.

Reply:
Use this test instead and it should work:

```php
// Test to make sure the callback has a non-null value.
(new Foo())->when(User::first(), function (Foo $foo, $user): void {
    assertType(User::class, $user);
});
```

Reply:
Thanks @erikgaal!

Reply:
@canvural could you have another look at this? Need your approval to run the tests